src/stdlib/distributions.jl

```diff
@@ -49,3 +49,49 @@ end
 function Distributions.logpdf(d::VecBinomialLogit{<:Real}, ks::Vector{<:Integer})
     return sum(logpdf_binomial_logit.(d.n, d.logitp, ks))
 end
+
+# OrderedLogistic distribution
+struct OrderedLogistic{T1, T2} <: DiscreteUnivariateDistribution
+   η::T1
+   cutpoints::Vector{T2}
+end
+
+# logpdf of the OrderedLogistic distribution
+function Distributions.logpdf(d::OrderedLogistic, k::Int)
+
+    K = length(d.cutpoints)+1
+
+    c =  d.cutpoints
+
+    if k==1
+        logp= log(logistic(c[k]-d.η))
+    elseif k<K
+        logp= log(logistic(c[k]-d.η) - logistic(c[k-1]-d.η))
+    else
+        logp= log(1-logistic(c[k-1]-d.η))
+    end
+
+    return(logp)
+end
+
+# rand of the OrderedLogistic distribution
+function Distributions.rand(d::OrderedLogistic)
+    cutpoints = d.cutpoints
+    η = d.η
+
+    if !issorted(cutpoints)
+        error("cutpoints are not sorted")
+    end
+
+    K = length(cutpoints)+1
+    c = vcat(-Inf, cutpoints, Inf)
+    l = [i for i in zip(c[1:(end-1)],c[2:end])]
```

Comment:
Could these two lines
```julia
l = [i for i in zip(c[1:(end-1)],c[2:end])]
ps = [logistic(η - l[i][1]) - logistic(η - l[i][2]) for i in 1:K]
```
be re-written as
```julia
ps = [logistic(η - i[1]) - logistic(η - i[2]) for i in zip(c[1:(end-1)],c[2:end])]
```
?

Doing this would avoid ever creating the intermediate array `l`, which would be nice.